### DISCUSSION TITLE
1Password SSH Agent

### DISCUSSION THREAD

**Discussion by belaarany** [dbeaver/dbeaver#19099] [Q&A]
Hi!

I'm trying to login to my database through SSH. I set "SSH Agent" as the authentication method, however it doesn't try to reach the SSH agent of 1Password. All I get is "exhausted available auth methods".

Has anyone made 1Password SSH Agent work with DBeaver?
What should I do to make DBeaver reach the 1P SSH agent?

**Comment by sebikeller** [comment]
Under macOS, I did not find a way to launch DBBeaver via Finder or Spotlight, so that I can use 1Password SSH Agent with DBeaver. 

But you can open a new shell and start DBeaver directly in the shell  by typing the command `env SSH_AUTH_SOCK=~/.1password/agent.sock /Applications/DBeaver.app/Contents/MacOS/dbeaver &` the 1Password Agent can be used by DBeaver.

You can also add `export SSH_AUTH_SOCK=~/.1password/agent.sock` (or `export SSH_AUTH_SOCK=~/Library/Group\ Containers/2BUA8C4S2C.com.1password/t/agent.sock` if you did not create a [symlink to the agent's socket](https://developer.1password.com/docs/ssh/get-started#step-4-configure-your-ssh-or-git-client)) to your shell config (e.g. `~/.zshrc` in case of ZSH). Then you can launch DBeaver like so `/Applications/DBeaver.app/Contents/MacOS/dbeaver &`.

**Reply by pmichelazzo** [reply]
Unfortunately your tip doesn't work. I've tried with the latest versions of both applications without success.

Any clue?

**Comment by sebikeller** [comment]
Did you select „Agent“ as authentication type in the advanced options of the SSH pane of the database connection properties window?

**Reply by pmichelazzo** [reply]
Yes, I changed to agent but unfortunately, doesn't work :(

**Reply by sebikeller** [reply]
I export the following two variables:
```
➜  ~ export | grep SSH                               
GIT_SSH=/usr/bin/ssh
SSH_AUTH_SOCK=/Users/<user>/.1password/agent.sock
```
and the start DBeaver from Terminal as described above and it works like a charm.

Maybe the [`GIT_SSH` variable](https://stackoverflow.com/a/13398657) is the missing piece or that I'm using "SSHJ" as SSH Implementation:
<img width="763" alt="DBeaver SSH configuration" src="https://user-images.githubusercontent.com/1693354/233945783-2fec2864-f54c-402d-b245-cf70fb456f89.png">

**Reply by pmichelazzo** [reply]
Yes, you're right. Both env variables was missing. Now it's working great.

The only sad part is to launch the DBeaver by terminal.

Thanks a lot!

**Comment by mopx** [comment]
@pmichelazzo you can use this one liner, it doesn't require you to keep the terminal window open.

export | grep SSH && open -a "dbeaver"

**Reply by pmichelazzo** [reply]
Hi @mopx 

Thanks for the tip.
Regards

**Reply by t4nz** [reply]
On Mac and Linux, 1Password explains how to configure the agent socket path using the SSH_AUTH_SOCK environment variable globally for every client

https://developer.1password.com/docs/ssh/agent/compatibility/#configure-ssh_auth_sock-globally-for-every-client

**Comment by anthonysomerset** [comment]
So i am also seeing this issue on linux, i largely see the same behaviour with the snap install or deb install on Ubuntu 24.04

If i launch the app normally it just won't see/use the 1password agent - even when i follow the 1password docs for enabling SSH_AUTH_SOCK globally

the only thing that works is setting SSH_AUTH_SOCK within my shell environment ( in my case i did it in the .zshrc file) and launching dbeaver from the terminal - this works every time with either SSH driver implementation

its almost like dbeaver is not loading profile.d or we need to do something additional to get dbeaver to set/use the environment variable nicely

**Comment by anthonysomerset** [comment]
i do have a possible workaround for linux  (deb install)

`cp /usr/share/applications/dbeaver-ce.desktop ~/.local/share/applications/`

then edit this however you prefer and update the `Exec` line to similar to the following - note that ~ expansion didn't work for me

```
Exec=env SSH_AUTH_SOCK=/home/user/.1password/agent.sock NO_AT_BRIDGE=1 /usr/share/dbeaver-ce/dbeaver %U
```

it should update instantly but occasionally you might need to run a refresh:

```
update-desktop-database ~/.local/share/applications
```
